### PR TITLE
Update openssl from 1.1.1l to 3.0.1

### PR DESCRIPTION
**Pull request by Lex-2008**
Ticket: ENT-8355

### CODE REVIEW

**Comment by Lex-2008** [comment]
@cf-bottom : build plz

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7638/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7638/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7638/

**Comment by Lex-2008** [comment]
Status: fails in C code: https://ci.cfengine.com/job/pr-pipeline/7679/
Can any of C devs have a look at this?

**Comment by vpodzime** [comment]
> Can any of C devs have a look at this?

There are many things that are different in OpenSSL 3.0: https://www.openssl.org/docs/manmaster/man7/migration_guide.html

**Comment by larsewi** [comment]
@cf-bottom may I haz a new build ?

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7777/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7777/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7777/

**Comment by larsewi** [comment]
@cf-bottom Jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7882/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7882/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7882/

File: deps-packaging/openssl/debian/rules [hunk 1]
```diff
@@ -36,7 +36,8 @@ build-stamp:
 	    --prefix=$(PREFIX) \
 	    $$HACK_FLAGS   \
 	    $$DEBUG_CFLAGS \
-	    $$LDFLAGS
+	    $$LDFLAGS \
+	    --libdir=lib
```
Comment (craigcomstock) [inline deps-packaging/openssl/debian/rules:40]: isn't `libdir` the destination for libraries? Seems odd to specify just a relative `lib` dir here.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7992/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7992/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7992/

**Comment by vpodzime** [comment]
> Sure, I triggered a build:
> 
> [Build Status](https://ci.cfengine.com//job/pr-pipeline/7992/)
> 
> **Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7992/

PHP fails to build with OpenSSL 3. @Lex-2008 can you check if we are at the latest version of PHP before we start patching it to make it build?

**Comment by Lex-2008** [comment]
bq. @Lex-2008 can you check if we are at the latest version of PHP before we start patching it to make it build?

you're right, as https://www.php.net/manual/en/openssl.requirements.php says:
bq. Only PHP >= 8.1 supports OpenSSL >= 3.0. 

but we had some issues with php 8.1.6 which I was hoping will be fixed later by php devs... I'll make a ticket for a C dev to look into.

**Comment by Lex-2008** [comment]
not needed anymore

**Pull request closed without merging**